src/components/modules/renderer.ts

```diff
@@ -44,11 +42,12 @@ export default class Renderer extends Module {
    * Make plugin blocks from array of plugin`s data
    *
    * @param {BlockToolData[]} blocks - blocks to render
+   * @param {boolean} readOnly - readOnly flag value
    */
-  public async render(blocks: BlockToolData[]): Promise<void> {
-    const chainData = blocks.map((block) => ({ function: (): Promise<void> => this.insertBlock(block) }));
+  public async render(blocks: BlockToolData[], readOnly: boolean): Promise<void> {
```

Comment:
why not to get the `readOnly` state from the public property of ReadOnly module? Then you won't need to pass `readOnly` in every method.